### DISCUSSION TITLE
Issue with Dismissing an alert pop up on an iOS devices on Browserstack

### DISCUSSION THREAD

**Discussion by troyyer** [webdriverio/webdriverio#12881] [Q&A]
Hi,

I'm trying to run a test on different iOS devices to dismiss an alert that has an ok and Cancel thru a cloud vendor Browserstack. The test works correctly on Desktop browsers including MacOS Safari and on Android mobile devices. On running the test on iOS devices the tests are failing due to the prompt being dismissed prior to the `await browser.dismissAlert();` command. the Prompts are always dismissed with the OK rather than the cancel prompt.
Please find my repo of the test
https://github.com/troyyer/wdiov8browserstackDismissAlertoniOSIssue
the error seen thru webdriverio is 

> no such alert: An attempt was made to operate on a modal dialog when one was not open
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

I am using the following test website
https://demo.automationtesting.in/Alerts.html
The Test website will mention that button you have clicked on the Alert Dialog box 
![image](https://github.com/webdriverio/webdriverio/assets/28956675/0a24851a-e347-4ce4-a1a7-c0efc31e4a47)

please see the following iPhone recording of the test execution
https://github.com/webdriverio/webdriverio/assets/28956675/2cee0f8b-b20e-4511-a8a7-992f737bbe5a
in the recording, you can see that the website shows that we have clicked the OK button
so to my uneducated understanding it seems like once the alert dialog is presented it somehow gets the OK button clicked to dismiss it.
here is the attached Browserstack Appium logs 
[Appium logs (1).txt](https://github.com/webdriverio/webdriverio/files/15328131/Appium.logs.1.txt)
Any debugging assistance on this would be appreciated

**Comment by christian-bromann** [comment]
Have you tried reaching out to the BrowserStack support? This seems like some issue with the combination of WebdriverIO + Appium + BrowserStack.
